in_toto/keylib.go

```diff
@@ -390,9 +389,12 @@ func GenerateSignature(signable []byte, key Key) (Signature, error) {
 		}
 		switch key.Scheme {
 		case rsassapsssha256Scheme:
-			hashed := sha256.Sum256(signable)
+			if _, ok := hashMapping["sha256"]; !ok {
+				return Signature{}, ErrUnsupportedHashAlgorithm
+			}
```

Comment:
We know that `"sha256"` is in `hashMapping`, right? I think we should trust our own code.

Reply:
Same comment applies to the rest of the diff where you check if an expected value is actually in `hashMapping`.

Reply:
Yes, I agree with you that this part was very defensive. I just wanted to make sure that this will run in a graceful error if somebody has ever the brilliant idea to remove sha256 support. :D

Reply:
I'm not a big fan of that kind of defensive programming. I think tests are a way better strategy to defend against such brilliant ideas. :)